### DISCUSSION TITLE
can't figure out how to create nested routes with route redirects.

### DISCUSSION THREAD

**Discussion by brucewongqbio** [TanStack/router#1996] [Q&A] [answered]
I'm trying to figure out how to render a route like this
`/house/garage/car` 

When I go to that route, I expect all pieces from each subpath to be rendered. So with the analogy, it should render a house with a garage and inside that garage is a car.

That's fine and all but I also want to prevent people from going to the `house/garage` path directly. If they try, it should redirect to `house/garage/car`. 

How can I get this working with file based routing? Sorry for the weird analogy but hope it makes sense and thank you!

**Comment by ViewableGravy** [comment, marked as answer]
One option you can consider is in the beforeLoad of your garage route, adding a redirect. for example

```ts
beforeLoad({ location }) {
  if (location.pathname === "/house/garage")
    throw redirect({ to: "/house/garage/car" })
}
```

Alternatively, if garage is simply an outlet, then you should probably show a 404 page at that route instead of having garage as a parent of car. (Note, I have not extensively used file based routing so my assumption here may be incorrect) but at the moment, it sounds like you have
- house
  - garage
    - car

When really you might want something more like
- house
    - garage/car

My understanding based on the [File-based routing docs](https://tanstack.com/router/latest/docs/framework/react/guide/file-based-routing#:~:text=.route.tsx%20File%20Type) is that you can do this using a folder structure like follows.

- house
  - garage
    - car.route.tsx

**Reply by brucewongqbio** [reply]
the garage acts as a template in my setup. Your before load suggestion worked like a charm ! Thank you.
